ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -458,6 +463,33 @@ run env@ChainDBEnv { varDB, .. } cmd =
     giveWithEq a =
       fmap (`WithEq` a) $ atomically $ stateTVar varNextId $ \i -> (i, succ i)
 
+-- | When the model is asked for the ledger DB, it reconstructs it by applying
+-- the blocks in the current chain, starting from the initial ledger state.
+-- Before the introduction of UTxO HD, this approach resulted in a ledger DB
+-- equivalent to the one maintained by the SUT. However, after UTxO HD, this is
+-- no longer the case since the ledger DB can be altered as the result of taking
+-- snapshots or opening the ledger DB (for instance when we process the
+-- 'WipeVolatileDB' command). Taking snapshots or opening the ledger DB cause
+-- the ledger DB to be flushed, which modifies its sequence of volatile and
+-- immutable states.
+--
+-- The model does not have information about when the flushes occur and it
+-- cannot infer that information in a reliable way since this depends on the low
+-- level details of operations such as opening the ledger DB. Therefore, we
+-- assume that the 'GetLedgerDB' command should return a flushed ledger DB, and
+-- we use this function to implement such command both in the SUT and in the
+-- model.
+--
+-- When we compare the SUT and model's ledger DBs, by flushing we are not
+-- comparing the immutable parts of the SUT and model's ledger DBs. However,
+-- this was already the case in before the introduction of UTxO HD: if the
+-- current chain contained more than K blocks, then the ledger states before the
+-- immutable tip were not compared by the 'GetLedgerDB' command.
+-- flush ::
+--      (LedgerSupportsProtocol blk)
+--   => DbChangelog.DbChangelog' blk -> DbChangelog.DbChangelog' blk
+-- flush = snd . DbChangelog.splitForFlushing
```

Comment:
Does this comment actually apply ATM given that the `GetLedgerDB` command is disabled?

Reply:
It does not apply currently, but it might if we decide to fix the tests. However, I'd argue that the proper way to handle this discrepancy between the model and the real impl is to change the `Eq` instance for the result of flushing. We do a similar thing with `IsValidResult`

Reply:
Captured this discussion in #1339

I think we can remove the commented out code for now